src/mswin/fg_joystick_mswin.c

```diff
@@ -129,27 +129,27 @@ static int fghJoystickGetOEMProductName ( SFG_Joystick* joy, char *buf, int buf_
         return 0;
 
     /* Open .. MediaResources\CurrentJoystickSettings */
-    _snprintf ( buffer, sizeof(buffer), "%s\\%s\\%s",
-                REGSTR_PATH_JOYCONFIG, joy->pJoystick.jsCaps.szRegKey,
-                REGSTR_KEY_JOYCURR );
+    _sntprintf ( buffer, _countof(buffer), _T("%s\\%s\\%s"),
+                 REGSTR_PATH_JOYCONFIG, joy->pJoystick.jsCaps.szRegKey,
+                 REGSTR_KEY_JOYCURR );
 
     lr = RegOpenKeyEx ( HKEY_LOCAL_MACHINE, buffer, 0, KEY_QUERY_VALUE, &hKey);
 
     if ( lr != ERROR_SUCCESS ) return 0;
 
     /* Get OEM Key name */
-    dwcb = sizeof(OEMKey);
+    dwcb = _countof(OEMKey);
```

Comment:
_countof does not exist on MSVC 6. From the docs it looks like it's simply a macro expanding to `sizeof foo / sizeof *foo`, so please use that instead.